PersonApi/V1/Gateways/DynamoDbGateway.cs

```diff
@@ -28,5 +29,17 @@ public async Task<Person> GetPersonByIdAsync(Guid id)
             var result = await _dynamoDbContext.LoadAsync<PersonDbEntity>(id).ConfigureAwait(false);
             return result?.ToDomain();
         }
+
+        [LogCall]
+        public async Task<Person> PostNewPersonAsync(PersonRequestObject requestObject)
+        {
+            _logger.LogDebug($"Calling IDynamoDBContext.SaveAsync");
+            var personDbEntity = requestObject.ToDatabase();
+            personDbEntity.Id = Guid.NewGuid();
```

Comment:
shouldn't this also be in the `ToDatabase()` method?

Reply:
Depends if we use the same request object (maybe renamed) for updating a person. 
But either way, I would say the factory `ToDatabase()` method should convert between the object types _only_, the Gateway method knows it is persisting a new object and should set a new id accordingly.

Reply:
Happy to go with either approach.

Reply:
I would expect that for the `Update` functionality, we'll need to add an Id into that object. Then in the `ToDatebase()` method it can conditionally check if it is set, or set it?